File: building-blocks/jpa/src/main/java/io/micro_spring_boot/building_blocks/jpa/domain/event/handlers/DomainEventHandler.java
package io.micro_spring_boot.building_blocks.jpa.domain.event.handlers;

import io.micro_spring_boot.building_blocks.jpa.domain.event.DomainEvent;
import org.springframework.transaction.event.TransactionPhase;
import org.springframework.transaction.event.TransactionalEventListener;

public interface DomainEventHandler<T extends DomainEvent> {
    @TransactionalEventListener(phase = TransactionPhase.BEFORE_COMMIT)
    void onApplicationEvent(T event);
}

File: building-blocks/jpa/src/main/kotlin/com/microspringboot/building_blocks/jpa/domain/event/handlers/DomainEventHandler.kt
package com.microspringboot.building_blocks.jpa.domain.event.handlers

import com.microspringboot.building_blocks.jpa.domain.event.DomainEvent
import org.springframework.transaction.event.TransactionPhase
import org.springframework.transaction.event.TransactionalEventListener

interface DomainEventHandler<T : DomainEvent> {
    @TransactionalEventListener(phase = TransactionPhase.BEFORE_COMMIT)
    fun onApplicationEvent(event: T)
}